Add repo-level documentation (#116)
elastic/synthetics-recorder: CONTRIBUTING.md

# Contributing

Contributions to this repository will be very welcome, both in the form of issues or pull-requests 😊

Before starting to work on a particular piece of code, please first open an issue to discuss the change before starting to work on it.

Please note that Elastic does have a [Code of Conduct](https://www.elastic.co/community/codeofconduct).

# Creating an issue

We expect contributors to fill our issue templates when creating new issues. It's important that contributors use those templates appropriately because doing so helps us triage issues more quickly, discuss them more efficiently, and respond to the issue's participants sooner.

> For bugs, for example, we expect contributors to provide clear instructions on how to reproduce the bug they've found, and, if possible a screenshot or GIF illustrating the problem. We also appreciate it when contributors can provide information with regards to what caused the bug to occur, even though that's not necessary for reporting a bug.

Reviewer: Super nit, just trying to pare off a couple of words.

```suggestion
For bugs, for example, we expect contributors to provide clear instructions on how to reproduce the bug they've found, and, if possible a screenshot or GIF illustrating the problem. We also appreciate it when contributors can provide information about the bug's cause, even though that's not necessary for reporting a bug.
```

Author: Great suggestion, I love it when we can make text more concise, thank you 💖

Once contributors create an issue, this project's maintainers will review and triage it, and we'll aim to get back to the issue's creator as soon as we can, although we can't promise any particular dates.

# Opening a pull-request

> Before submitting your PR, ensure it complies with the specs in the issue you're solving, and run both the tests and linting tasks specified in `package.json`. These will run in CI anyway, so you'll have to have these tasks passing before you can merge your pull-request.

Reviewer: Super nit again.

```suggestion
Before submitting your PR, ensure it complies with the specs in the issue you're solving, and run both the tests and linting tasks specified in `package.json`. These are required by CI, so they need to be passing before you can merge your pull-request.
```

When opening a pull-request, make sure to specify which issue it solves, so that the PR and issue get linked. Do the same in your commit messages by adding the issue number to it.

Just like when opening an issue, it's important to fill all the sections in our pull-request templates appropriately, so that we can review your contribution sooner, and provide actionable feedback in less time.

> After your issue is opened **and CI is passing*, the team will do its best to review it and get back to you as soon as we can, although we can't promise dates given the team may have other priorities.

Reviewer: Super super nit. This sounds a little nicer IMO.

```suggestion
After your issue is opened **and CI is passing*, the team will do its best to review it and get back to you as soon as we can, although we can't commit to a specific timeframe.
```

> Please note that you'll have to sign [Elastic's CLA to be able to contribute code to this repo](https://www.elastic.co/contributor-agreement).

Reviewer 2: If we can extend the list by adding Workflow, Testing docs it would be great. We can also do it as a followup.

Author: That's a great idea! For now I think it makes sense to merge these and then adding Workflow/Testing docs given we're still iterating on the E2E stuff.
